src/main/java/org/oasis_eu/portal/front/my/rest/OrganizationImportController.java

```diff
@@ -0,0 +1,176 @@
+package org.oasis_eu.portal.front.my.rest;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.oasis_eu.portal.core.mongo.model.geo.GeographicalArea;
+import org.oasis_eu.portal.front.generic.BaseController;
+import org.oasis_eu.portal.services.PortalSystemUserService;
+import org.oasis_eu.portal.services.dc.geoarea.GeographicalAreaService;
+import org.oasis_eu.portal.services.dc.organization.DCOrganization;
+import org.oasis_eu.portal.services.dc.organization.DCOrganizationService;
+import org.oasis_eu.portal.services.dc.organization.OrganizationService;
+import org.oasis_eu.portal.services.kernel.UserMembershipService;
+import org.oasis_eu.portal.ui.UIOrganization;
+import org.oasis_eu.spring.datacore.DatacoreClient;
+import org.oasis_eu.spring.datacore.model.DCResource;
+import org.oasis_eu.spring.kernel.exception.WrongQueryException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import org.thymeleaf.util.StringUtils;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
+@RestController
+@RequestMapping("/my/api/organization/import")
+class OrganizationImportController extends BaseController {
+
+    private static final Logger logger = LoggerFactory.getLogger(OrganizationImportController.class);
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private DatacoreClient datacore;
+
+    @Autowired
+    private OrganizationService organizationService;
+
+    @Autowired
+    PortalSystemUserService portalSystemUserService;
+
+    @Autowired
+    UserMembershipService userMembershipService;
+
+    @Autowired
+    DCOrganizationService dcOrganizationService;
+
+    @Autowired
+    private GeographicalAreaService geographicalAreaService;
+
+    @Value("${application.dcOrg.project: org_0}")
+    private String dcOrgProjectName;// = "org_0";
+
+    @Value("${application.importPassword}")
+    private String importPassword;
+
+    @PostMapping
+    public ResponseEntity<String> organizationsImport(@RequestHeader String password,
```

Comment:
Why not using a standard HTTP authentication method ?

Reply:
Since it was just for a single specific WS, I figured it was simple like that than setting up a multiple security through all the app

Reply:
It is roughly the same effort and you'll be more "standardy"